src/service.rs

```diff
@@ -0,0 +1,297 @@
+use std::ffi::OsString;
+use std::path::PathBuf;
+
+use lazy_static::lazy_static;
+use structopt::StructOpt;
+
+use crate::cli::{handle_config_errors, read_configs};
+use crate::config_paths;
+
+lazy_static! {
+    pub static ref DEFAULT_SERVICE_NAME: &'static str = built_info::PKG_NAME;
+}
+
+#[derive(StructOpt, Debug)]
+#[structopt(rename_all = "kebab-case")]
+pub struct Opts {
+    #[structopt(subcommand)]
+    sub_command: Option<SubCommand>,
+}
+
+#[derive(StructOpt, Debug)]
+#[structopt(rename_all = "kebab-case")]
+struct InstallOpts {
+    /// The name of the service to install.
+    #[structopt(long)]
+    name: Option<String>,
+
+    /// The display name to be used by interface programs to identify the service like Windows Services App
+    #[structopt(long)]
+    display_name: Option<String>,
+
+    /// The configuration files that will be used by the service.
+    /// If no configuration file is specified, will target default configuration file.
+    #[structopt(name = "config", short, long)]
+    config_paths: Vec<PathBuf>,
```

Comment:
I initially wanted to reuse the same `--config` parameter from the `RootOpts` but failed to do so.

```
release\vector.exe --config C:\Users\oktal\tmp\vector.toml -- service install
error: The subcommand 'service' wasn't recognized
        Did you mean 'service'?

If you believe you received this message in error, try re-running with 'vector.exe -- service'

USAGE:
    vector.exe --config <config>...

For more information try --help
```

Reply:
I wonder if it's worth us making it global (https://kbknapp.github.io/clap-rs/clap/struct.Arg.html#method.global) in the future...

Either way, I think this is fine for now.